### ISSUE TITLE
Crash on adding first waypoint to a cache

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#13873]
### Describe your problem!

c:geo crashed when adding a waypoint to a cache that does not have a waypoint yet.

### How to reproduce?

- open a cache without waypoints
- swipe to waypoint tab
- tap "add waypoint"

### Actual result after these steps?

c:geo crashes (system popup "c:geo stopped"), but recovers into `EditWaypointActivity`

### Expected result after these steps?

prevent crash

### Reproducible

Yes

### c:geo Version

current nightly

### Additional Information

```
 Process: cgeo.geocaching, PID: 17117
java.lang.ArrayIndexOutOfBoundsException: length=0; index=-1
at java.util.ArrayList.get(ArrayList.java:439)
at cgeo.geocaching.CacheDetailActivity$WaypointsViewCreator.lambda$setContent$2$cgeo-geocaching-CacheDetailActivity$WaypointsViewCreator(CacheDetailActivity.java:2119)
at cgeo.geocaching.CacheDetailActivity$WaypointsViewCreator$$ExternalSyntheticLambda10.onClick(Unknown Source:6)
at android.view.View.performClick(View.java:7506)
at com.google.android.material.button.MaterialButton.performClick(MaterialButton.java:1194)
at android.view.View.performClickInternal(View.java:7483)
at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
at android.view.View$PerformClick.run(View.java:29334)
at android.os.Handler.handleCallback(Handler.java:942)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7872)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

**Comment by eddiemuc** [comment]
I noted this too and "side-fixed" it in PR #13867

**Comment by eddiemuc** [comment]
With PR #13867 merged, we can close this issue
